src/main/resources/native/macosx/AppDelegate.m

```diff
@@ -28,9 +28,7 @@
 #import <Cocoa/Cocoa.h>
 #include <pthread.h>
 
-extern void *run_main(int argc, char* argv[]);
-
-const char* args[] = {"myapp"};
```

Comment:
We are removing `myapp`, which seems not needed at all. It is still present in the iOS and Android args, though. But maybe for another issue.

Reply:
While launching the application, when the main() method gets invoked, the first argument that is being passed is always the name of the executable that is being run. I don't think Graal does anything specific with that argument. What I have done in this PR is to basically replace `"myapp"` with `argv[0]`.

I have no idea how iOS and android are being launched by the OS. In any case, you can't launch the application from command line and pass in arbitrary command line arguments on those systems anyway, so the issue that is being fixed by this PR doesn't apply there.